### DISCUSSION TITLE
How to find out the number of bits read from a stream or an array of bits?

### DISCUSSION THREAD

**Discussion by Foxiks** [scott-griffiths/bitstring#252] [Q&A] [answered]
Hello! I would like to know, how could I find out the number of bits read from a file using constbitstraem or BitArray?
example:
```
import bitstring
x = 1
y = 2
with open('data.bin', 'rb') as file:
    stream = bitstring.constbitstream(file)
stream.read(int(x)).bin
stream.read(int(y)).bin
print(bitstring.totalreadbits(stream))
>> 3
```

**Comment by scott-griffiths** [comment, marked as answer]
There is a `pos` attribute on bitstrings that support reading, which gives the current bit position. So you can look at `stream.pos` before and after a set of reads to see how much has been read. This isn't supported for the `Bits` or `BitArray` classes as you can't treat them like a stream.

So in short you can just use `stream.pos` to get a bit position.

**Reply by Foxiks** [reply]
Oh, thanks!
